flyio/caveats.go

```diff
@@ -314,3 +316,66 @@ func (c *NoAdminFeatures) Prohibits(a macaroon.Access) error {
 
 	return nil
 }
+
+// Commands is a list of command names allowed by this token.
+type Commands struct {
+	Commands []string `json:"commands"`
+}
+
+func init()                                         { macaroon.RegisterCaveatType(&Commands{}) }
+func (c *Commands) CaveatType() macaroon.CaveatType { return CavCommands }
+func (c *Commands) Name() string                    { return "Commands" }
+
+func (c *Commands) Prohibits(a macaroon.Access) error {
+	f, isFlyioAccess := a.(CommandGetter)
+	if !isFlyioAccess {
+		return fmt.Errorf("%w: access isnt CommandGetter", macaroon.ErrInvalidAccess)
+	}
+
+	commandArgs := f.GetCommand()
+	if commandArgs == nil {
+		return fmt.Errorf("%w: only authorized for command execution", resset.ErrResourceUnspecified)
+	}
+
+	commandName := commandArgs[0]
+	found := slices.Contains(c.Commands, commandName)
```

Comment:
In `resset`, there's semantic meaning to specifying the zero value of an ID type: I'm requiring that this resource type is specified in the access, but I'm not limiting which IDs can be acted on. Do we want `&Commands{""}` to have any special meaning? Totally fine if not.

Reply:
In the updated code, an empty Commands caveat is "reject all", and a Commands caveat with a zero value Command in it is "allow all".